pubsub/api/Pubsub.Samples/ExactlyOnceDeliverySubscriberAsync.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright 2022 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START pubsub_subscriber_exactly_once]
+
+using Google.Cloud.PubSub.V1;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Google.Cloud.PubSub.V1.SubscriberClient;
+
+public class ExactlyOnceDeliverySubscriberAsyncSample
+{
+    public async Task<IEnumerable<string>> ExactlyOnceDeliverySubscriberAsync(string projectId, string subscriptionId)
+    {
+        // subscriptionId should be an exactly once delivery subscription.
+        SubscriptionName subscriptionName = SubscriptionName.FromProjectSubscription(projectId, subscriptionId);
+        SubscriberClient subscriber = await SubscriberClient.CreateAsync(subscriptionName);
+        // To get the status of ACKnowledge (ACK) or Not ACKnowledge (NACK) request in exactly once delivery subscriptions,
+        // create a subscription handler that inherits from Google.Cloud.PubSub.V1.SubscriptionHandler. 
+        // For more information see Google.Cloud.PubSub.V1.SubscriptionHandler reference docs here:
+        // TODO: [add link to reference docs when we have them].
+        var subscriptionHandler = new SampleSubscriptionHandler();
+        Task subscriptionTask = subscriber.StartAsync(subscriptionHandler);
+        // The subscriber will be running until it is stopped.
+        await Task.Delay(5000);
+        await subscriber.StopAsync(CancellationToken.None);
+        // Let's make sure that the start task finished successfully after the call to stop.
+        await subscriptionTask;
+        return subscriptionHandler.SuccessfulAckedIds;
+    }
+
+    // Sample handler to handle messages and ACK/NACK responses.
+    public class SampleSubscriptionHandler : SubscriptionHandler
+    {
+        public ConcurrentBag<string> SuccessfulAckedIds { get; } = new ConcurrentBag<string>();
+
+        public override async Task<Reply> HandleMessage(PubsubMessage message, CancellationToken cancellationToken)
+        {
+            string text = System.Text.Encoding.UTF8.GetString(message.Data.ToArray());
+            Console.WriteLine($"Message {message.MessageId}: {text}");
+            return await Task.FromResult(Reply.Ack);
+        }
+
+        // This method will receive the response for every ACKnowledge request that is successful or failed.
```

Comment:
Maybe in summary tags?
And consider doing the same for the HandleMessage method.